data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessor.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@DataPrepperPlugin(name = "obfuscate", pluginType = Processor.class, pluginConfigurationType = ObfuscationProcessorConfig.class)
+public class ObfuscationProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    private static final String COMMON_PATTERN_REGEX = "^%\\{([A-Z_]+)}$";
```

Comment:
Is this the approach we want to take for using common patterns - `%{}`?

It indicates that we could have multiple side-by-side.

e.g.

```
%{EMAIL_ADDRESS}%{PHONE_NUMBER_US}
```

Reply:
The regex does not allow multiple ones side by side.  It must start with `%{` and end with `}` and has only allow `A-Z_` in between. So %{EMAIL_ADDRESS}%{PHONE_NUMBER_US} will not treat as a common patten, it will throw an error as well as it's invalid (Not a valid regex string).

In terms of format, Any other thoughts?  This `%{}` is similar to Grok pattern.